server/models/schema.sql

```diff
@@ -2561,4 +2573,20 @@ CREATE TABLE `configuration_analysis_tools` (
   CONSTRAINT `config_analysis_tools__analysis_tool_type` FOREIGN KEY (`analysis_tool_type_id`) REFERENCES `analysis_tool_type` (`id`)
 ) ENGINE=InnoDB DEFAULT CHARACTER SET = utf8mb4 DEFAULT COLLATE = utf8mb4_unicode_ci;
 
+
+DROP TABLE IF EXISTS `cost_center_aggregate`;
+CREATE TABLE `cost_center_aggregate` (
+  `period_id`       MEDIUMINT(8) UNSIGNED NOT NULL,
```

Comment:
What "periods" are these?  Will we have the flexibility to do aggregates over months, quarters, years, and arbitrary date ranges?  I see in the Bhima interface that some places 'periods' refer to months of a fiscal year.   While months may be adequate, I could easily imagine wanting cost/revenue aggregates over more arbitrary periods including quarters, years, and the immediately preceeding 6 months, etc.

Reply:
You are correct - period is a month.  The goal of these tables (period_total, for example) is to reduce the number of records we need to sum over.  But we can still sum for greater periods than just a month.  If you need finer-grained resolution (e.g. you are mid-month), you must do the summing yourself on the general_ledger.  Some of this logic is shown in the [accounts/extra.js](https://github.com/IMA-WorldHealth/bhima/blob/39c32b0522f335ebcded814b70d69c660524a446/server/controllers/finance/accounts/extra.js#L123) file.

Reply:
Ok.